dbms/src/Storages/Transaction/PartitionStreams.cpp

```diff
@@ -167,7 +170,14 @@ static void writeRegionDataToStorage(
     /// Try read then write once.
     {
         if (atomic_read_write(false))
+        {
+            LOG_FMT_INFO(log, "[for test only] first decode sucess");
+
+            fiu_do_on(FailPoints::force_pause_query_until_write_finish, {
+                FailPointHelper::disableFailPoint(FailPoints::pause_query_until_write_finish);
+            });
```

Comment:
You can use a handy helper SCOPE_EXIT to avoid writing multiple times for exit points:

```cpp
SCOPE_EXIT({
    fiu_do_on(FailPoints::force_pause_query_until_write_finish, {
        FailPointHelper::disableFailPoint(FailPoints::pause_query_until_write_finish);
    });
});
```